AdvanceDS/Generalizedist/AdvanceList.java

```diff
@@ -0,0 +1,261 @@
+		int count=0;
+		int openBrac=0;
+		int closeBrac=0;
+		int comma=0;
+		int countElement=0;
+		currentChar=input.charAt(count);
+		while(count < input.length()) {
+			switch (currentChar) {
+			case '(':
+				openBrac++;
+				count++;
+				if(count < input.length()) {
+					currentChar=input.charAt(count);
+				}
+				break;
+			case ')':
+				closeBrac++;
+				count++;
+				if(count < input.length()) {
+					currentChar=input.charAt(count);
+				}
+				break;
+			case ',':
+				comma++;
+				count++;
+				if(count < input.length()) {
+					currentChar=input.charAt(count);
+				}
+				break;
+			case '0':
+			case '1':
+			case '2':
+			case '3':
+			case '4':
+			case '5':
+			case '6':
+			case '7':
+			case '8':
+			case '9':
+				count++;
+				currentChar=input.charAt(count);
+				while(currentChar > '0' && currentChar < '9') {
+					count++;
+					currentChar=input.charAt(count);
+				}
+				countElement++;
+				break;
+			default:
+				throw new MyException(e);
+			}
+		}
+		if(countElement-1 != comma) {
+			return false;
+		}
+		if(closeBrac != openBrac) {
+			return false;
+		}
+		else {
+			return true;
+		}
+	}
+	//method to create a list if list is valid
+	private Node createList(String input) {
+		currentChar = input.charAt(index);
+		Node head = null;
+		Node previous = null;
+		while(index < input.length()-1) {
+			switch (currentChar) {
+			case '(':
+				index++;
+				if(index < input.length()) {
+					currentChar=input.charAt(index);
+				}
+				Node newNode = new Node();
+				newNode.setNestedList(true);
+				newNode.setNext(null);
+				newNode.setData(createList(input));
+				if(head == null) {
+					head = newNode;
+					previous = newNode;
+				}
+				else {
+					previous.setNext(newNode);
+					previous = newNode;
+				}
+				break;
+			case ',':
+				index++;
+				currentChar = input.charAt(index);
+				break;
+				//case for take integer values from list
+			case '0':
+			case '1':
+			case '2':
+			case '3':
+			case '4':
+			case '5':
+			case '6':
+			case '7':
+			case '8':
+			case '9':
+				String number = Character.toString(currentChar);
+				index++;
+				currentChar=input.charAt(index);
+				while(currentChar > '0' && currentChar < '9') {
+					number += currentChar;
+					index++;
+					currentChar=input.charAt(index);
+				}
+				//to convert number string in integer
+				int data = Integer.parseInt(number);
+				//create a new node with tag false
+				Node newNode1 = new Node();
+				newNode1.setNestedList(false);
+				newNode1.setData(data);
+				newNode1.setNext(null);
+				if(head == null) {
+					head = newNode1;
+					previous = newNode1;
+				}
+				else {
+					previous.setNext(newNode1);
+					previous = newNode1;
+				}
+				break;
+			case ')': 
+				index++;
+				currentChar = input.charAt(index);
+				break;
+			default :
+				break;
+			}
+		}
+		return head;
+	}
+	//method to get maximum value from entered list
+	private int maxHelper(Node start) {
+		if(start == null) {
+			return max;
```

Comment:
Generally,static variables in the same category as global variables, and prefer to avoid them when you can. In particular, using static variables to do recursion can lead to problems. You can use method arguments and method should be returning something, by that you can eliminate use of global scoped variable.